lib/Test/Pod/CoverageChange.pm

```diff
@@ -0,0 +1,246 @@
+package Test::Pod::CoverageChange;
+# ABSTRACT: check perl modules against their pod coverage
+
+=pod
+
+=head1 NAME
+
+Pod coverage calculator test file.
+
+=head1 SYNOPSIS
+
+    use Test::Pod::CoverageChange qw(check);
+
+    Test::Pod::CoverageChange::check('lib', {
+        Module::With::3::expected::naked::subs              => 3,
+        AnotherModule::With::10::expected::naked::subs      => 10,
+        YetAnotherModule::With::1::expected::naked::subs    => 1,
+        YetAnotherModule::With::5::expected::naked::subs    => 5,
+    }, [
+        We::Ignore::ThisModule,
+        We::Also::Ignore::This::Module
+    ]);
+
+=head1 DESCRIPTION
+
+It checks all files that placed under a given directory against their
+POD coverage to see if all existing subs have POD and also
+POD syntax to see if is there an POD syntax error or not?
+
+Prints the percentage of POD coverage in B<TODO> test format for the packages that we allowed to have naked subs.
+Prints an error message if our latest changes increased/decreased numbers of naked sub for the packages that we allowed to have naked sub.
+Prints an error message if a naked allowed package has 100% POD coverage. (We should remove it from the C<%naked_packages> variable list.)
+Ignores to check every package that we pass as C<@ignored_package>
+Prints a proper message for the newly added packages.
+
+Prints C<ok> for the files that have no POD syntax error.
```

Comment:
This is inaccurate - this shouldn't be *printing* anything, it's running tests. `prove -v` happens to render this as raw TAP output, but there are many other test harnesses and formatters.